### PR TITLE
Fix failing tests in the python ecosystem

### PR DESCRIPTION
**Pull request by amazimbe**
### What are you trying to accomplish?

3 tests are failing on CI in the python ecosystem.

Failed examples:

rspec ./spec/dependabot/python/file_parser/setup_file_parser_spec.rb:178 # Dependabot::Python::FileParser::SetupFileParser for setup.cfg parse length is expected to eq 15
rspec ./spec/dependabot/python/file_parser/setup_file_parser_spec.rb:219 # Dependabot::Python::FileParser::SetupFileParser for setup.cfg parse a tests_require dependencies has the right details
rspec ./spec/dependabot/python/file_parser_spec.rb:999 # Dependabot::Python::FileParser parse with a setup.cfg length is expected to eq 15

It looks like we are running a python script in a subprocess to parse a setup.cfg file here:
https://github.com/dependabot/dependabot-core/blob/main/python/helpers/run.py#L13 .

I'm not sure if something has changed in the python parser but for some reason it's
not including the dependencies under tests_require and that's causing some assertions
in the tests to fail.

The dependencies missing / being ignored are here:
https://github.com/dependabot/dependabot-core/blob/main/python/spec/fixtures/setup_files/setup_with_requires.cfg#L25

https://github.com/dependabot/dependabot-core/issues/10388

### How will you know you've accomplished your goal?
Tests will be green again

### Checklist

- [x] I have run the complete test suite to ensure all tests and linters pass.
- [x] I have thoroughly tested my code changes to ensure they work as expected, including adding additional tests for new functionality.
- [x] I have written clear and descriptive commit messages.
- [x] I have provided a detailed description of the changes in the pull request, including the problem it addresses, how it fixes the problem, and any relevant details about the implementation.
- [x] I have ensured that the code is well-documented and easy to understand.

### CODE REVIEW

**Review by jurre — CHANGES_REQUESTED** [review]
It looks like this just updates the tests to reflect that those dependencies aren't being included?

Let's instead figure out what caused this to change, determine if it is a regression and fix it if so.

**Comment by amazimbe** [comment]
> It looks like this just updates the tests to reflect that those dependencies aren't being included?
> 
> Let's instead figure out what caused this to change, determine if it is a regression and fix it if so.

I agree though my concern was that we'll be blocked from merging and shipping.

**Comment by honeyankit** [comment]
> ./spec/dependabot/python/file_parser/setup_file_parser_spec.rb:178

@amazimbe : The issue is that the two dependencies (`pytest==2.9.1`, `responses==0.5.1`) are not present in the parsed output after the method [parse_setup](https://github.com/dependabot/dependabot-core/blob/main/python/helpers/lib/parser.py#L132) in the native helper (`parser.py`) is executed. 

The code takes all the dependencies from the fixture [setup_with_requires.cfg](https://github.com/dependabot/dependabot-core/blob/main/python/spec/fixtures/setup_files/setup_with_requires.cfg) which is 15 and write to temp setup.cfg which is correct but when the native helper parse the temp setup.cfg at this [place](https://github.com/dependabot/dependabot-core/blob/main/python/lib/dependabot/python/file_parser/setup_file_parser.rb#L69) there only 13 dependencies and two of the above stated dependencies is missing.

**Comment by honeyankit** [comment]
> > ./spec/dependabot/python/file_parser/setup_file_parser_spec.rb:178
> 
> @amazimbe : The issue is that the two dependencies (`pytest==2.9.1`, `responses==0.5.1`) are not present in the parsed output after the method [parse_setup](https://github.com/dependabot/dependabot-core/blob/main/python/helpers/lib/parser.py#L132) in the native helper (`parser.py`) is executed.
> 
> The code takes all the dependencies from the fixture [setup_with_requires.cfg](https://github.com/dependabot/dependabot-core/blob/main/python/spec/fixtures/setup_files/setup_with_requires.cfg) which is 15 and write to temp setup.cfg which is correct but when the native helper parse the temp setup.cfg at this [place](https://github.com/dependabot/dependabot-core/blob/main/python/lib/dependabot/python/file_parser/setup_file_parser.rb#L69) there only 13 dependencies and two of the above stated dependencies is missing.

@amazimbe I should had seen your initial description. You already got this.

File: python/helpers/lib/parser.py [hunk 1]
```diff
@@ -226,25 +226,27 @@ def fake_open(*args, **kwargs):
...
                 parse_requirements(requires, req_type, setup_cfg)
 
-            extras_require = config.get("options", {}).get(
-                "extras_require", {}
-            )
-            for key, value in extras_require.items():
-                parse_requirements(
-                    value, "extras_require:{}".format(key), setup_cfg
-                )
+            if config.has_section('options.extras_require'):
```
Comment (honeyankit) [inline python/helpers/lib/parser.py:241]: @amazimbe : I took the liberty to make changes in your branch, hope thats ok. Executing native helper script provides below warning where `setuptools.config.read_configuration` is going to be deprecated so we have to anyways replace it in the future. The setuptools currently getting used is [v72.1.0](https://github.com/pypa/setuptools/releases/tag/v72.1.0) which was released 2 weeks ago and this is the issue. I have changed the native helper to use `config  parser` to to parse the setup.cfg keeping all the functionality same.

```bash
how to update the setuptools.config.read_configuration to use build.util.project_wheel_metadata for reading setup.cfg provided here. update the parser.py

/opt/python/lib/parser.py:229: SetuptoolsDeprecationWarning: Deprecated API usage.
!!

        ********************************************************************************
        As setuptools moves its configuration towards `pyproject.toml`,
        `setuptools.config.read_configuration` became deprecated.

        For the time being, you can use the `setuptools.config.setupcfg` module
        to access a backward compatible API, but this module is provisional
        and might be removed in the future.

        To read project metadata, consider using
        ``build.util.project_wheel_metadata`` (https://pypi.org/project/build/).
        For simple scenarios, you can also try parsing the file directly
        with the help of ``configparser``.
        ********************************************************************************
```

Comment (honeyankit) [reply python/helpers/lib/parser.py:244]: @jurre ^^

File: python/spec/dependabot/python/file_parser/setup_file_parser_spec.rb [hunk 2]
```diff
@@ -216,7 +216,7 @@
       describe "a tests_require dependencies" do
         subject(:dependency) { dependencies.find { |d| d.name == "responses" } }
 
-        it "has the right details" do
+        it "has the right details", skip: "Issue: https://github.com/dependabot/dependabot-core/issues/10388" do
```
Comment (abdulapopoola) [inline python/spec/dependabot/python/file_parser/setup_file_parser_spec.rb]: @amazimbe , can we revert these changes (based on @honeyankit 's suggestions) so that the tests are still accurate?

Comment (honeyankit) [reply python/spec/dependabot/python/file_parser/setup_file_parser_spec.rb]: I have already reversed all the test case changes

File: python/helpers/lib/parser.py (hunk 1 above)
Comment (amazimbe) [reply python/helpers/lib/parser.py:244]: @honeyankit Yes, that is fine.

**Review by jurre — APPROVED** [review]
Thanks @honeyankit for following up, that seems to address the underlying issue 🙇
